### PR TITLE
chore: add various trigger conditions callouts

### PR DESCRIPTION
**Pull request by cjbell**
### Description

Docs for trigger conditions

### Tasks

KNO-5873

### CODE REVIEW

File: content/designing-workflows/step-conditions.mdx [hunk 1]
```diff
@@ -26,6 +27,16 @@ See our [guide on the Knock conditions model](/send-and-manage-data/conditions)
...
 
+## Types of step conditions
+
+### Trigger step conditions
+
+A [trigger step](/designing-workflows/overview#the-trigger-step#the-trigger-step) can have one or more step conditions, which will be evaluated on the trigger of the workflow for the recipient. When the conditions evaluate to false then the workflow **will be halted** and no other steps will be executed.
+
+### Other step conditions
+
+For all function and channel steps, step conditions will be evaluated when the step is executed. If the conditions on the step evaluate to false, then the step will be **skipped** and the subsequent step will be invoked, or the workflow will terminate.
```
Comment (connorlindsey) [inline content/designing-workflows/step-conditions.mdx:38]: What determines if the workflow terminates or the step is only skipped?

File: content/mapi.mdx [hunk 2]
```diff
@@ -177,6 +177,11 @@ You can retrieve, update, or create a workflow as well as list all workflows in
     type="WorkflowStep[]"
     description="A list of workflow step objects in the workflow, which may contain any of: channel step, delay step, batch step, fetch step."
   />
+  <Attribute
+    name="conditions"
+    type="object"
+    description="A conditions object that describes one or more conditions to be met for the workflow to be executed."
```
Comment (cellomatt) [inline content/mapi.mdx:183]: Maybe either "one or more trigger conditions" or "for the workflow to be triggered" could tie this back to the fact that we're calling these "trigger conditions"?

Comment (cellomatt) [reply content/mapi.mdx:183]: "trigger step conditions"? 🤔 maybe this is fine how it is 🤷

**Review by cellomatt — APPROVED** [review]
+1 to Connor's clarifying question, other than that I think this is all super clear 🚀

**Pull request merged**